packages/headless/src/features/history/history-slice.ts

```diff
@@ -47,29 +47,64 @@ const isEqual = (current: HistoryState, next: HistoryState) => {
 const isContextEqual = (current: ContextState, next: ContextState) =>
   JSON.stringify(current.contextValues) === JSON.stringify(next.contextValues);
 
-// TODO: compare all facets non-idle values, comparing current values changes history after 1st request
-const isFacetsEqual = (current: FacetSetState, next: FacetSetState) =>
-  JSON.stringify(current) === JSON.stringify(next);
+type FacetStateWithCurrentValues = Record<
+  string,
+  CurrentValues<BaseFacetValueRequest>
+>;
 
-const isDateFacetsEqual = (
-  current: DateFacetSetState,
-  next: DateFacetSetState
-) => JSON.stringify(current) === JSON.stringify(next);
+const isFacetsEqual = (
+  current: FacetStateWithCurrentValues,
+  next: FacetStateWithCurrentValues
+) => {
+  for (const [key, value] of Object.entries(next)) {
+    if (!current[key]) {
+      return false;
+    }
 
-const isNumericFacetsEqual = (
-  current: NumericFacetSetState,
-  next: NumericFacetSetState
-) => JSON.stringify(current) === JSON.stringify(next);
+    const currentSelectedValues = current[key].currentValues.filter(
+      (value) => value.state === 'selected'
+    );
+    const nextSelectedValues = value.currentValues.filter(
+      (value) => value.state === 'selected'
+    );
+
+    if (
+      JSON.stringify(currentSelectedValues) !==
+      JSON.stringify(nextSelectedValues)
+    ) {
+      return false;
+    }
+  }
+
+  return true;
+};
 
 const isCategoryFacetsEqual = (
   current: CategoryFacetSetState,
   next: CategoryFacetSetState
-) => JSON.stringify(current) === JSON.stringify(next);
+) => {
+  for (const [key, value] of Object.entries(next)) {
+    if (!current[key]) {
+      return false;
+    }
 
-const isFacetOptionsEqual = (
-  current: FacetOptionsState,
-  next: FacetOptionsState
-) => JSON.stringify(current) === JSON.stringify(next);
+    const currentSelectedValues = partitionIntoParentsAndValues(
+      current[key]?.request.currentValues
+    ).parents;
+    const nextSelectedValues = partitionIntoParentsAndValues(
```

Comment:
I think there is an inconsistency here with the other facets. I believe with the category facet, show more/show less will trigger a snapshot, whereas with the other three facets it will not? I believe the `retrieveCount` is set on the last parent.

Reply:
Interesting, I'll filter out some of the request parameters